src/GShark/Geometry/Point3.cs

```diff
@@ -717,7 +717,37 @@ public int InPolygon(Polygon polygon)
             }
             return inside ? 1 : -1;
         }
-        
+
+
+        // Function to find mid point from cloud 
+        public static Point3 AveragePoint(List<Point3> points)
+        {
+            Point3 pointTotSum = Point3.Origin;
+            for (int i = 0; i < points.Count(); i++)
+            {
+                pointTotSum += points[i];
+            }
+            Point3 mid = pointTotSum / points.Count();
+            return mid;
+        }
+
+        // Function that finds point in a cloud of points to a target point
+        public Point3 CloudClosestPoint(IEnumerable<Point3> cloud)
```

Comment:
This method as a reversed pattern then what we usually do. If you think of `mesh.ClosestPoint(testPoint)` or `plane.ClosestPoint(testPoint)`, then this one should be `cloud.ClosestPoint(testPoint)`.

Ok, we don't have a cloud object, but what we are doing here is something like `testPoint.ClosestPoint(cloud)` which is not intuitive.

Make the function static, and place both the cloud and the testpoint as function parameter: `Point3.CloudClosestPoint(cloud, testPoint)`.

Also, usually the `ClosestPoint` function is returning a new point, instead here we are returning the same object within the cloud without copying it. It's good that way because of the use we do of the function, but we need to write it in the function docstring. Do this into the `<remark></remark>` tags.